src/nwnsdk/postgres/postgres_client.py

```diff
@@ -17,41 +17,38 @@ def __init__(self, postgres_config: PostgresConfig):
         initialize_db("nwn", postgres_config)
 
     def send_input(
-        self, job_id: uuid4, job_name: str, work_flow_type: WorkFlowType, esdl_str: str, user_name: str
+            self, job_id: uuid4, job_name: str, work_flow_type: WorkFlowType, esdl_str: str, user_name: str
     ) -> None:
         with session_scope() as session:
             new_job = Job(
                 job_id=job_id,
                 job_name=job_name,
-                work_flow_type=work_flow_type.value,
-                map_editor_user=user_name,
-                status="registered",
+                work_flow_type=work_flow_type,
+                user_name=user_name,
+                status=JobStatus.REGISTERED,
                 input_esdl=esdl_str,
                 added_at=datetime.now(),
             )
             session.add(new_job)
 
     def retrieve_input_esdl(self, job_id: uuid4) -> str:
-        session: Session
         LOGGER.debug("Retrieving esdl for job %s", job_id)
         with session_scope() as session:
-            stmnt = select(Job.input_esdl).where(Job.job_id == job_id)
-            input_esdl: str = session.scalar(stmnt)
-        return input_esdl
+            stmnt = select(Job).where(Job.job_id.is_(job_id))
```

Comment:
I don't necessarily understand why yet, but the `.is_()` does not work. It requires the `==`. 

Example log of the error you otherwise receive:
```
optimization_worker_1  | 2023-08-18 07:20:46,440 [MainThread][main.py:76][INFO]: Received message to work on job c4fea0de-0b24-4ae8-a7b0-7601caad9f31
optimization_worker_1  | 2023-08-18 07:20:46,482 INFO sqlalchemy.engine.Engine select pg_catalog.version()
optimization_worker_1  | 2023-08-18 07:20:46,482 INFO sqlalchemy.engine.Engine [raw sql] {}
optimization_worker_1  | 2023-08-18 07:20:46,486 INFO sqlalchemy.engine.Engine select current_schema()
optimization_worker_1  | 2023-08-18 07:20:46,487 INFO sqlalchemy.engine.Engine [raw sql] {}
optimization_worker_1  | 2023-08-18 07:20:46,490 INFO sqlalchemy.engine.Engine show standard_conforming_strings
optimization_worker_1  | 2023-08-18 07:20:46,491 INFO sqlalchemy.engine.Engine [raw sql] {}
optimization_worker_1  | 2023-08-18 07:20:46,496 INFO sqlalchemy.engine.Engine BEGIN (implicit)
optimization_worker_1  | 2023-08-18 07:20:46,512 INFO sqlalchemy.engine.Engine SELECT job.job_id, job.job_name, job.work_flow_type, job.user_name, job.status, job.input_config, job.input_esdl, job.output_esdl, job.added_at, job.running_at, job.stopped_at, job.logs 
optimization_worker_1  | FROM job 
optimization_worker_1  | WHERE job.job_id IS %(job_id_1)s::UUID
optimization_worker_1  | 2023-08-18 07:20:46,513 INFO sqlalchemy.engine.Engine [generated in 0.00088s] {'job_id_1': UUID('c4fea0de-0b24-4ae8-a7b0-7601caad9f31')}
optimization_worker_1  | 2023-08-18 07:20:46,515 INFO sqlalchemy.engine.Engine ROLLBACK
optimization_worker_1  | Traceback (most recent call last):
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1965, in _exec_single_context
optimization_worker_1  |     self.dialect.do_execute(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 921, in do_execute
optimization_worker_1  |     cursor.execute(statement, parameters)
optimization_worker_1  | psycopg2.errors.SyntaxError: syntax error at or near "'c4fea0de-0b24-4ae8-a7b0-7601caad9f31'"
optimization_worker_1  | LINE 3: WHERE job.job_id IS 'c4fea0de-0b24-4ae8-a7b0-7601caad9f31'::...
optimization_worker_1  |                             ^
optimization_worker_1  | 
optimization_worker_1  | 
optimization_worker_1  | The above exception was the direct cause of the following exception:
optimization_worker_1  | 
optimization_worker_1  | Traceback (most recent call last):
optimization_worker_1  |   File "/usr/local/lib/python3.8/runpy.py", line 194, in _run_module_as_main
optimization_worker_1  |     return _run_code(code, main_globals, None,
optimization_worker_1  |   File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
optimization_worker_1  |     exec(code, run_globals)
optimization_worker_1  |   File "/app/optimizer_worker/main.py", line 132, in <module>
optimization_worker_1  |     main()
optimization_worker_1  |   File "/app/optimizer_worker/main.py", line 128, in main
optimization_worker_1  |     OptimizationWorker().work()
optimization_worker_1  |   File "/app/optimizer_worker/main.py", line 84, in work
optimization_worker_1  |     self.nwn_client.broker_client.wait_for_data({Queue.StartWorkflowOptimizer: self.on_message})
optimization_worker_1  |   File "/app/nwnsdk/rabbitmq/rabbitmq_client.py", line 71, in wait_for_data
optimization_worker_1  |     self.channel.start_consuming()
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 1883, in start_consuming
optimization_worker_1  |     self._process_data_events(time_limit=None)
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 2044, in _process_data_events
optimization_worker_1  |     self.connection.process_data_events(time_limit=time_limit)
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 851, in process_data_events
optimization_worker_1  |     self._dispatch_channel_events()
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 567, in _dispatch_channel_events
optimization_worker_1  |     impl_channel._get_cookie()._dispatch_events()
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/pika/adapters/blocking_connection.py", line 1510, in _dispatch_events
optimization_worker_1  |     consumer_info.on_message_callback(self, evt.method,
optimization_worker_1  |   File "/app/optimizer_worker/main.py", line 77, in on_message
optimization_worker_1  |     input_esdl_string = self.nwn_client.db_client.retrieve_input_esdl(job_id)
optimization_worker_1  |   File "/app/nwnsdk/postgres/postgres_client.py", line 39, in retrieve_input_esdl
optimization_worker_1  |     job: Job = session.scalar(stmnt)
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2296, in scalar
optimization_worker_1  |     return self._execute_internal(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/session.py", line 2141, in _execute_internal
optimization_worker_1  |     result: Result[Any] = compile_state_cls.orm_execute_statement(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/orm/context.py", line 293, in orm_execute_statement
optimization_worker_1  |     result = conn.execute(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1412, in execute
optimization_worker_1  |     return meth(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 483, in _execute_on_connection
optimization_worker_1  |     return connection._execute_clauseelement(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1635, in _execute_clauseelement
optimization_worker_1  |     ret = self._execute_context(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1844, in _execute_context
optimization_worker_1  |     return self._exec_single_context(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1984, in _exec_single_context
optimization_worker_1  |     self._handle_dbapi_exception(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2339, in _handle_dbapi_exception
optimization_worker_1  |     raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1965, in _exec_single_context
optimization_worker_1  |     self.dialect.do_execute(
optimization_worker_1  |   File "/usr/local/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 921, in do_execute
optimization_worker_1  |     cursor.execute(statement, parameters)
optimization_worker_1  | sqlalchemy.exc.ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "'c4fea0de-0b24-4ae8-a7b0-7601caad9f31'"
optimization_worker_1  | LINE 3: WHERE job.job_id IS 'c4fea0de-0b24-4ae8-a7b0-7601caad9f31'::...
optimization_worker_1  |                             ^
optimization_worker_1  | 
optimization_worker_1  | [SQL: SELECT job.job_id, job.job_name, job.work_flow_type, job.user_name, job.status, job.input_config, job.input_esdl, job.output_esdl, job.added_at, job.running_at, job.stopped_at, job.logs 
optimization_worker_1  | FROM job 
optimization_worker_1  | WHERE job.job_id IS %(job_id_1)s::UUID]
optimization_worker_1  | [parameters: {'job_id_1': UUID('c4fea0de-0b24-4ae8-a7b0-7601caad9f31')}]
optimization_worker_1  | (Background on this error at: https://sqlalche.me/e/20/f405)
```

Reply:
Edit: Just checked the resulting SQL. The `==` operator is converted to the SQL `=` operator while the `is_()` function is converted to the SQL `IS` keyword. Seems the difference is similar to the Pythonic `is` and `==` where `is` checks not equality but sameness: https://www.postgresql.org/docs/15/sql-syntax-lexical.html#SQL-PRECEDENCE